src/util/memory.cpp

```diff
@@ -57,4 +74,73 @@ AlignedChunk getPageAlignedChunk(long offset, long length)
 
     return c;
 }
+
+// -------------------------
+// Dirty page tracking
+// -------------------------
+
+void resetDirtyTracking()
+{
+    FILE* fd = fopen(CLEAR_REFS, "w");
+    if (fd == nullptr) {
+        throw std::runtime_error("Could not open clear_refs");
+    }
+
+    // Write 4 to the file to track from now on
+    // https://www.kernel.org/doc/html/v5.4/admin-guide/mm/soft-dirty.html
+    char value[] = "4";
+    size_t nWritten = fwrite(value, sizeof(char), 1, fd);
+    if (nWritten != 1) {
+        throw std::runtime_error("Failed to write to clear_refs");
```

Comment:
Both the exceptions in here will only happen if there's a permissions error on the file itself, so it will only happen if we've misconfigured an environment, and this exception will be triggered from both the main thread and child threads. A log line could be useful though.